06-building-a-program-start-to-finish/solving-the-problem/output/inverted.js

```diff
@@ -0,0 +1,231 @@
+    }
+    try {
+      jsonContent.forEach((doc) => {
+        if (!doc.title || !doc.text) {
+          throw formatError;
+        } else {
+          isValid = {
+            status: true,
+            jsonContent
+          };
+          return isValid;
+        }
+      });
+    } catch (error) {
+      if (error.name === 'validate file structure') return false;
+      throw error;
+    }
+    return isValid;
+  }
+  /**
+   *
+   *
+   * @static
+   * @param {Array} words -takes in an array of words with possible duplicates.
+   * @returns{Array} item -returns an array of words without duplicates.
+   *
+   * @memberOf InvertedIndex
+   */
+  static uniqueWords(words) {
+    if (Array.isArray(words)) {
+      const checked = {};
+      return words.filter((item) => {
+        if (!checked[item]) {
+          checked[item] = true;
+          return item;
+        }
+        return null;
+      });
+    }
+    return ['invalid data type supplied'];
+  }
+
+  /**
+   *
+   *
+   * @static
+   * @param {String} text - a string of spaced words with unwanted characters
+   * @returns{String} text - a string of spaced words with no unwanted characters
+   *
+   * @memberOf InvertedIndex
+   */
+  static tokenizeWords(text) {
+    const invalid = /[.,/#!$%^&*;:{}=\-_`~()]/g;
+    text = text.replace(invalid, '');
+    return text;
+  }
+  /**
+   *
+   *
+   * @static
+   * @param {Object} docObject - an object containing a string of characters.
+   * @returns{Object} docObject - an object containing an array of words.
+   *
+   * @memberOf InvertedIndex
+   */
+  static splitAndSort(docObject) {
+    const words = docObject.toLowerCase().split(' ').sort();
+    docObject = InvertedIndex.uniqueWords(words);
+    return docObject;
+  }
+
+  /**
+   *
+   *
+   * @static
+   * @param {Object} jContent - single object of a JSON formatted file
+   * @returns{Object} concatenatedText - returns the concatenated values of object keys
+   *
+   * @memberOf InvertedIndex
+   */
+  static concatenateText(jContent) {
+    // if a document exists? combine title and text to split at once
+    let concatenatedText = {};
+    concatenatedText = `${jContent.title} ${jContent.text}`;
+    return concatenatedText;
+  }
+
+  /**
+   *
+   *
+   * @param {Array} book - contents of a JSON formatted file
+   * @param {String} filename - name of the file
+   * @returns{Object} this.indexedFiles - contains filenames and their corresponding contents.
+   *
+   * @memberOf InvertedIndex
+   */
+  createIndex(book, filename) {
+    const indices = {};
+    const splittedWords = {};
+    book.forEach((doc, key) => {
+      const joinedkeys = InvertedIndex.concatenateText(doc);
+      const tokenizedWords = InvertedIndex.tokenizeWords(joinedkeys);
+      splittedWords[key] = InvertedIndex.splitAndSort(tokenizedWords);
+    });
+    Object.keys(splittedWords).forEach((keys) => {
+      splittedWords[keys].forEach((words) => {
+        if (!indices.hasOwnProperty(words)) {
```

Comment:
I believe there is a better way to call `Object.prototype` methods directly. Please [see link](https://github.com/airbnb/javascript#objects--prototype-builtins)